packages/plugin-testing-integration/src/app.js

```diff
@@ -61,47 +62,40 @@ async function initImaApp(bootConfigMethods = {}) {
   /**
    * Initializes JSDOM environment for the application run
    */
-  function _initJSDom() {
-    /**
-     * Copies object props from src to target
-     *
-     * @param {object} src
-     * @param {object} target
-     */
-    function copyProps(src, target) {
-      Object.defineProperties(target, {
-        ...Object.getOwnPropertyDescriptors(src),
-        ...Object.getOwnPropertyDescriptors(target),
-      });
-    }
+  async function _initJSDom() {
+    const content = await _getIMAResponseContent();
 
-    const jsdom = new JSDOM(
-      `<!doctype html><html><body><div id="${config.masterElementId}"></div></body></html>`,
-      {
-        pretendToBeVisual: true,
-        url: `${config.protocol}//${config.host}/`,
-      }
-    );
+    // SPA jsdom interpreter
+    const jsdom = new JSDOM(content, {
+      pretendToBeVisual: true,
+      url: `${config.protocol}//${config.host}/`,
+    });
+
+    // Setup node environment to work with jsdom window
     const { window } = jsdom;
 
     global.window = window;
-    global.document = window.document;
-    global.navigator = {
-      userAgent: 'node.js',
-    };
-    copyProps(window, global);
     global.jsdom = jsdom;
-    global.$IMA = global.$IMA || {};
-    global.window.$IMA = global.$IMA;
-    global.window.$Debug = global.$Debug;
-    global.window.scrollTo = () => {};
-    global.window.fetch = global.fetch;
 
-    global.$IMA.$Protocol = config.protocol;
-    global.$IMA.$Host = config.host;
-    global.$IMA.$Env = config.environment;
-    global.$IMA.$App = config.$App || {};
+    // Call all page scripts (jsdom build-in runScript creates new V8 context, unable to mix with node context)
+    const pageScripts = jsdom.window.document.getElementsByTagName('script');
+    if (typeof config.pageScriptEvalFn === 'function') {
+      for (const script of pageScripts) {
+        config.pageScriptEvalFn(script, global, window, { ...config });
```

Comment:
Both `global` and `window` are available globally, does it make any sense to pass them through the function?